base/compiler/ssair/inlining.jl

```diff
@@ -1039,107 +1166,21 @@ function assemble_inline_todo!(ir::IRCode, sv::OptimizationState)
         nu = countunionsplit(sig.atypes)
         if nu == 1 || nu > sv.params.MAX_UNION_SPLITTING
             if !isa(info, MethodMatchInfo)
-                info = nothing
+                info = recompute_method_matches(sig.atype, sv)
             end
             infos = Any[info]
-            splits = Any[sig.atype]
         else
             if !isa(info, UnionSplitInfo)
-                splits = Any[]
+                infos = MethodMatchInfo[]
                 for union_sig in UnionSplitSignature(sig.atypes)
-                    push!(splits, argtypes_to_type(union_sig))
+                    push!(infos, recompute_method_matches(union_sig, sv))
                 end
-                infos = Any[nothing for i = 1:length(splits)]
             else
-                splits = info.sigs
                 infos = info.matches
             end
         end
 
-        cases = Pair{Any, Any}[]
-        signature_union = Union{}
-        only_method = nothing  # keep track of whether there is one matching method
-        too_many = false
-        local meth
-        local fully_covered = true
-        for i in 1:length(splits)
-            atype = splits[i]
-            info = infos[i]
-            if info === nothing
-                info = recompute_method_matches(atype, sv)
-            end
-            meth = info.applicable
-            if meth === false || info.ambig
-                # Too many applicable methods
-                # Or there is a (partial?) ambiguity
```

Comment:
What happened to this?

Reply:
Looks like it got lost in rebase. Will fix.